### DISCUSSION TITLE
Hillshading from DEM files

### DISCUSSION THREAD

**Discussion by JFritzle** [devemux86/cruiser#71] [Q&A] [answered]
To continue previous discussion [Hill Shading from DEM files](https://groups.google.com/g/cruiser-users/c/Wyn_3JLo8ds):

What about using one's own _local_ hillshading server?
[Mapsforge tile server](https://github.com/telemaxx/mapsforgesrv) can be set up as an alpha-transparent hillshading tiles server, using Mapsforge rendering engine for rendering tiles on the fly from locally stored DEM files. Hillshading tiles can overlay map tiles.

![x](https://user-images.githubusercontent.com/62614244/223724235-75108460-8161-42ca-8422-6ef471284b26.jpg)

Server was started by
`java -jar <jar file folder>/mapsforgesrv4java8.jar -p <port number> -m "" -hs simple(0.1,1.666) -hm 2. -d <dem files folder>`
 
Tiles were retrieved by
`http://127.0.0.1:<port number>/<zoom level>/<x tile number>/<y tile number>.png`

**Comment by devemux86** [comment]
Thanks for the suggestion. Do you mean:

- DEM source with srtm hgt files
Available only in Mapsforge, for example [HillshadingMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/HillshadingMapViewer.java).
- Or semi-transparent png tiles with [shading (levels of gray)](https://wiki.openstreetmap.org/wiki/Hillshading)
Can be used in Mapsforge + VTM. Is there any provider available?

**Reply by JFritzle** [reply]
I mean semi-transparent png tiles with [shading (levels of gray)](https://wiki.openstreetmap.org/wiki/Hillshading).

Example:
Semi-transparent PNG tile with levels of gray for zoom level = 13, x tile number = 4253, y tile number = 2913
![Hillshading_tile_13_4253_2913](https://user-images.githubusercontent.com/62614244/223747827-82dd909e-da83-488e-b2f2-6319f34e948e.png)

Hillshading tile can be used as an overlay for any rendered map tile with same zoom level, x tile number, y tile number.
Map will shine through depending on slope angle:
slope angle 0 degrees -> transparency = 100 %, slope angle 90 degrees -> transparency = 0 %.

Locally installed tiles provider is this [Mapsforge tile server](https://github.com/telemaxx/mapsforgesrv) using publicly available, previously downloaded and locally stored DEM files (3" or 1" HGT files).

**Reply by devemux86** [reply]
Cruiser already supports semi-transparent hillshade png tiles, it had this feature before.
Now there doesn't seem to be any server available, so hillshading is currently disabled.

The idea of custom map servers in Cruiser was suggested before and I have it in my list.
Then we could add a raster tile server with a url: `https://server/${z}/${x}/${y}.png`

**Reply by JFritzle** [reply]
Custom map servers as mentioned were perfect!
Maybe, even tiles/maps from different enabled sources could finally get composed.

**Reply by devemux86** [reply]
> Maybe, even tiles/maps from different enabled sources could finally get composed.

Indeed, in the final implementation we could combine multiple map sources.

**Comment by devemux86** [comment, marked as answer]
Hillshading is available again in [Cruiser 3.0.18](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Map | Layers | Hillshading" or search settings for "hill".
